middleware/plugin/middleware.go

```diff
@@ -0,0 +1,164 @@
+//go:build plugin
+
+package plugin
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"net/http"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	"github.com/Southclaws/fault"
+	"github.com/rs/zerolog/log"
+	genPlugin "github.com/siemens/wfx/generated/plugin"
+	"github.com/siemens/wfx/generated/plugin/client"
+	"github.com/siemens/wfx/middleware/logging"
+)
+
+type MW struct {
+	plugin    Plugin
+	chMessage chan Message
+	quitChan  chan error
+}
+
+func NewMiddleware(plugin Plugin, quitChan chan error) (*MW, error) {
+	contextLogger := log.With().Str("plugin", plugin.Name()).Logger()
+	contextLogger.Debug().Msg("Creating new plugin middleware")
+	chMessage, err := plugin.Start(quitChan)
+	if err != nil {
+		_ = plugin.Stop()
+		return nil, fault.Wrap(err)
+	}
+	return &MW{
+		plugin:    plugin,
+		chMessage: chMessage,
+		quitChan:  quitChan,
+	}, nil
+}
+
+func (mw *MW) Wrap(next http.Handler) http.Handler {
+	var cookieCounter atomic.Uint64
```

Comment:
I assume the overflow behavior is wrapping? Shouldn't be a problem as we don't have so much requests/responses in-flight...

Reply:
Yes, the alternative would be to use something like UUIDs (which is slightly more expensive to exchange between wfx and the plugin).